msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerMsalController.java

```diff
@@ -81,21 +89,23 @@ public AcquireTokenResult acquireToken(AcquireTokenOperationParameters parameter
         //Wait to be notified of the result being returned... we could add a timeout here if we want to
         final Bundle resultBundle = mBrokerResultFuture.get();
 
-        return getAcquireTokenResult(resultBundle);
+        // For MSA Accounts Broker doesn't save the accounts, instead it just passes the result along,
+        // MSAL needs to save this account locally for future token calls.
+        saveMsaAccountToCache(resultBundle, (MsalOAuth2TokenCache)parameters.getTokenCache());
 
+        return getAcquireTokenResult(resultBundle);
```

Comment:
If the call to `save()` returns `false` for any reason, we should think what smart thing we can do here to signal there was an issue

Reply:
Might be worth adding a return value to `saveMsaToCache()`